### ISSUE TITLE
Legend loses gradient if shared between layers

### ISSUE THREAD

**Issue by dechamps** [vega/vega-lite#6258]
The following:

```json
{
  "layer": [{
    "mark": { "type": "line" },
    "encoding": {
      "color": {
        "type": "quantitative",
        "field": "c"
      }
    }
  }, {
    "mark": { "type": "circle" },
    "encoding": {
      "color": {
        "type": "quantitative",
        "field": "c"
      }
    }
  }],
  "data": {"values": [{"c": 1}, {"c": 2}]}
}
```

[Currently produces](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEA2CGCeBTATpAXGA2qCFIFtZkBrdMYKAF3gAdEyYBLAO3rAF8AacXSRZgMYB7ACYsA5mRy48w6ENQZpMvNToMAjgFdYzSo0qx9AN3rcVvAGaNE0EQwGQeM9s448u5NwSKklVWnoMSAFGZAFoNi5vfmExZkklN1kheUUvC1VAzR09AyNGU0hzTMhrW3tgx2T3Fw8AXRLIESNYKUhjWGgtRABnMmwQsgBGT2AhjAAmdnrXVyA):

![visualization(50)](https://user-images.githubusercontent.com/866859/78450456-936f8800-7676-11ea-8836-067c99c1f10f.png)

This is surprising - I would have expected a gradient legend.

If [a layer is removed](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEA2CGCeBTATpAXGA2qCFIFtZkBrdMYKAF3gAdEzIBjAS2UenrAF8AacXSIgB2jAPYATZkIDmZHLjxjoo1BnkK81OgwCOAV1hDKzSrGMA3enw0CAZs0TRxDRpH4Ku77vy4Bda1DiZrBykOaw0HqIAM5k2ExkAIy85AkYAEx+np5AA) (which one is removed doesn't matter), the output is correct:

![visualization(52)](https://user-images.githubusercontent.com/866859/78450497-d03b7f00-7676-11ea-9c79-b9252dd815a2.png)

**Comment by dechamps** [comment]
Up until vega-lite 4.8.0, I [used to be able to work around this problem](https://jsfiddle.net/2rjysedn/3/) by setting `legend` to `null` in one of the layers. But sadly, that workaround [doesn't work any more](https://jsfiddle.net/2rjysedn/4/) because of #6259.

**Comment by domoritz** [comment]
The issue here is that we are adding symbolType although we should not be. It's probably a bug in the legend merge logic.

<img width="639" alt="Screen Shot 2020-04-04 at 11 56 38" src="https://user-images.githubusercontent.com/589034/78459079-5a7de600-766b-11ea-8c6f-42b951174e1e.png">

**Comment by domoritz** [comment]
We can force the legend type to be gradient. I wonder where we merge incorrectly 🤔 

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "layer": [
    {
      "mark": {"type": "line"},
      "encoding": {
        "color": {
          "type": "quantitative",
          "field": "c",
          "legend": {"type": "gradient"}
        }
      }
    },
    {
      "mark": {"type": "circle"},
      "encoding": {"color": {"type": "quantitative", "field": "c"}}
    }
  ],
  "data": {"values": [{"c": 1}, {"c": 2}]}
}
```

**Comment by dechamps** [comment]
Yep, that workaround works with my chart. Thank you! Should have tried simply setting the `type`, seems obvious in retrospect :| I must have missed it in the legend property list.

Of course it would be even better if people didn't have to set the type explicitly, so I'll keep this open to fix the default behaviour.

**Comment by domoritz** [comment]
Completely agree. I'm just noting what I noticed in my debugging so far. We should fix the behavior.

**Comment by domoritz** [comment]
Okay, so the issue is that Vega uses a different legend when both fill and stroke are encoding a scale. That's a bit surprising but also kinda makes sense since only a symbol can encode both gradient and fill. 

Removing one or the other does not make sense in the general case so I think that the behavior is actually as expected.

**Comment by kanitw** [comment]
I don't think this is expected behavior

I think we could fix this in a few ways:

1) in Vega (when fill and stroke encodes the same color). 
or 2) encode "color" in legend parse and convert it to fill or stroke later in assemble.  

That said, since there is a good workaround, I'll make this issue P3.

**Comment by domoritz** [comment]
Yes, Vega could make a different decision about showing data when only stroke and fill are encoded and they are encoded with the same scale. I also agree that it's low priority to change.
